cabal.project

```diff
@@ -53,3 +53,33 @@ package nothunks
 -- See the note on nix/pkgs/default.nix:agdaPackages for why this is here.
 -- (NOTE this will change to ieee754 in newer versions of nixpkgs).
 extra-packages: ieee, filemanip
+
+if impl(ghc >= 9.6)
+  allow-newer:
+    , *:base
+    , *:ghc-prim
+    , *:mtl
+    , *:template-haskell
+    , *:transformers
+    , constraints-extras:*
+    , dependent-map:*
+    , hedgehog:*
+    , protolude:*
+    , th-compat:*
+
+  constraints:
+    , microlens >= 0.2.0.0
```

Comment:
Please add a comment explaining these (I would expect us cabal to pick the compatible versions of these libraries automatically) and when we can remove them.